src/main/java/org/broadinstitute/hellbender/metrics/InsertSizeMetricsCollector.java

```diff
@@ -53,135 +69,154 @@ protected InsertSizeMetricsCollectorArgs makeArg(SAMRecord samRecord, ReferenceS
-    public final class PerUnitInsertSizeMetricsCollector implements PerUnitMetricCollector<InsertSizeMetrics, Integer, InsertSizeMetricsCollectorArgs> {
-        final EnumMap<SamPairUtil.PairOrientation, Histogram<Integer>> Histograms = new EnumMap<>(SamPairUtil.PairOrientation.class);
-        final String sample;
-        final String library;
-        final String readGroup;
-        private double totalInserts = 0;
-
-        public PerUnitInsertSizeMetricsCollector(final String sample, final String library, final String readGroup) {
-            this.sample = sample;
-            this.library = library;
-            this.readGroup = readGroup;
-            String prefix = null;
-            if (this.readGroup != null) {
-                prefix = this.readGroup + ".";
-            }
-            else if (this.library != null) {
-                prefix = this.library + ".";
-            }
-            else if (this.sample != null) {
-                prefix = this.sample + ".";
+    /**
+     * Combine two InsertSizeMetricsCollector objects and return a single InsertSizeMetricsCollector
+     * object representing the combined results. NOTE: this implementation is destructive in that it
+     * merges the source into the target and returns the merged target as the combined object.
+     * @param target target destination of merged metrics
+     * @param source source of metrics to be merged into target
+     * @return single object representing the combined source and target objects
+     */
+    public InsertSizeMetricsCollector combine(InsertSizeMetricsCollector target, InsertSizeMetricsCollector source) {
+        target.combine(source);
+        return target;
+    }
+
+    /**
+     * Combine two PerUnitInsertSizeMetricsCollector objects and return a single PerUnitInsertSizeMetricsCollector
+     * object representing the combined results.
+     * @param collector1 source PerUnitInsertSizeMetricsCollector
+     * @param collector2 target PerUnitInsertSizeMetricsCollector
+     * @return single PerUnitInsertSizeMetricsCollector object representing the combined source and target objects
+     */
+    public PerUnitInsertSizeMetricsCollector combineUnit(
+            PerUnitInsertSizeMetricsCollector collector1,
+            PerUnitInsertSizeMetricsCollector collector2) {
+        return collector1.combine(collector2);
+    }
+
+    /**
+     * Finish the metrics collection by saving any results to a metrics file.
+     * @param metricsFile a metricsFile where the collected metrics should be stored
+     * @param inputName the name of the input, for optional inclusion in the metrics file
+     * @param authHolder
+     */
+    public void finish(
+            final MetricsFile<InsertSizeMetrics, Integer> metricsFile,
+            final String inputName,
+            final AuthHolder authHolder)
+    {
+        super.finish();
```

Comment:
this is a strange call. Usually, methods only call super if they want to invoke the method that is overriden by the current method. Here, the call is to a **different** method hierarchy. Can this simply become a virtual call to `finish()` rather than a non-virtual hardwired call to `MultiLevelCollectorReducible.finish()` ?